docs/source/design/progress_reporting.rst

```diff
@@ -0,0 +1,178 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_progress_reporting:
+
+******************
+Progress reporting
+******************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current implementation.
+
+
+Progress reporting is implemented via the logging system. A dedicated function
+:py:func:`datalad.log.log_progress` represents the main API for progress
+reporting. For some standard use cases, the utilities
+:py:func:`datalad.log.with_progress` and
+:py:func:`datalad.log.with_result_progress` can simplify result reporting
+further.
+
+
+Design and implementation
+=========================
+
+This basic idea is to use an instance of datalad's loggers to emit log messages
+with particular attributes that are picked up by
+:py:class:`datalad.log.ProgressHandler`, and are acted on differently,
+depending on configuration and conditions of a session (e.g., interactive
+terminal sessions vs.  non-interactive usage in scripts). This variable
+behavior is implemented via the use of log filters and handlers.  Roughly
```

Comment:
```suggestion
behavior is implemented via the use of :py:mod:`logging` standard library log filters and handlers.  Roughly
```
not sure if that is the right syntax but wanted to add explicitly that it is not some our "log filters" machinery, but from the standard library

Reply:
Done.